pkg/action/diff.go

```diff
@@ -164,55 +183,98 @@ func fileDestination(ctx context.Context, c bincapz.Config, fr, tr *bincapz.File
 	}
 }
 
-type diffReports struct {
-	Added     string
-	AddedFR   *bincapz.FileReport
-	Removed   string
-	RemovedFR *bincapz.FileReport
+// filterMap filters orderedmap pairs by checking for matches against a slice of compiled regular expression patterns.
+func filterMap(om *orderedmap.OrderedMap[string, *bincapz.FileReport], ps []*regexp.Regexp, c chan<- *orderedmap.Pair[string, *bincapz.FileReport], g *errgroup.Group) {
```

Comment:
This is where we really pare down what we need to iterate through when calculating the Levenshtein distance and merging the two reports.